src/layer/LayersFactory.ts

```diff
@@ -244,19 +244,21 @@ export class LayersFactory {
     overrideConstructorParams: Record<string, any> | null,
     reqConfig: RequestConfiguration,
   ): Promise<AbstractLayer[]> {
-    const parsedXml = await fetchGetCapabilitiesXml(baseUrl, reqConfig);
-    const layersInfos = parsedXml.WMS_Capabilities.Capability[0].Layer[0].Layer.map(layerInfo => ({
-      layerId: layerInfo.Name[0],
-      title: layerInfo.Title[0],
-      description: layerInfo.Abstract ? layerInfo.Abstract[0] : null,
-      dataset: null,
-      legendUrl:
-        layerInfo.Style && layerInfo.Style[0].LegendURL
-          ? layerInfo.Style[0].LegendURL[0].OnlineResource[0]['$']['xlink:href']
-          : layerInfo.Layer && layerInfo.Layer[0].Style && layerInfo.Layer[0].Style[0].LegendURL
-          ? layerInfo.Layer[0].Style[0].LegendURL[0].OnlineResource[0]['$']['xlink:href']
-          : null,
-    }));
+    const parsedLayers = await fetchLayersFromGetCapabilitiesXml(baseUrl, reqConfig);
+    const layersInfos = parsedLayers
+      .filter(layerInfo => layerInfo.Name)
```

Comment:
layers without name are already removed in `fetchLayersFromGetCapabilitiesXml`.fftr

Reply:
Thank you, fixed!